### DISCUSSION TITLE
Does Autogluon Have to Load the Model from Disk for Every Prediction Call?

### DISCUSSION THREAD

**Discussion by barmanroys** [autogluon/autogluon#2303] [Q&A] [answered]
Using Autogluon (version 0.3.1) on Python 3.8, Ubuntu 20.04 for tabular regression. This is the code sample. 

```
model=TabularPredictor(label=TARGET_COL, verbosity=4).fit(train_data=pd.concat(objs=[X_train, y_train], verify_integrity=True, axis=1),   time_limit=60*.5)
predictions=model.predict(data=pred_features) # Each call produces a detailed log of loading the file
```

The thing is, each call of the second line produces the following log on console. 

```
Loading: AutogluonModels/ag-20221104_084020/models/WeightedEnsemble_L2/model.pkl
Loading: AutogluonModels/ag-20221104_084020/models/ExtraTreesMSE/model.pkl
Loading: AutogluonModels/ag-20221104_084020/models/LightGBMXT/model.pkl
Loading: AutogluonModels/ag-20221104_084020/models/NeuralNetFastAI/model.pkl
Loading: AutogluonModels/ag-20221104_084020/models/NeuralNetFastAI/model-internals.pkl
Loading: AutogluonModels/ag-20221104_084020/models/RandomForestMSE/model.pkl
```

which suggests each call is loading the model from disk, when apparently it has no reason to, as the model is being trained as part of the same process (`.py `file). If it is indeed loading all the models from drive for each `.predict()` call, is not that incredibly slow? And is there any option to use the model directly from memory when it is available? In fact, the prediction with gluon is indeed slow compared to benchmarks like xgboost or catboost, and I suspect this is the reason. 

If it is not being loaded from disk, then how to interpret the log message?

**Comment by gradientsky** [comment, marked as answer]
You can [keep models in memory](https://auto.gluon.ai/0.3.1/tutorials/tabular_prediction/tabular-indepth.html#keeping-models-in-memory) via calling `predictor.persist_models()`.
